Standardized OpenAI config format and updated documentation issue #1180

anuja12mishra opened the pull request:
## Description
This PR addresses the issue of inconsistent AI service configuration by:
1. Making OpenAI configuration consistent with other AI services.
2. Adding documentation for configuring AI services.
3. Ensuring backward compatibility for the legacy OpenAI configuration format.

## Changes
- Updated OpenAI configuration to use the `services` block and `apiKey`.
- Added new documentation under `src/backend/src/modules/puterai/doc/ai-services-config.md`.
- Linked the new documentation from the general configuration docs.

## Related Issue
Fixes #1180

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/HeyPuter/puter?pullRequest=1195) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/HeyPuter/puter?pullRequest=1195) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/HeyPuter/puter?pullRequest=1195) it.</sub>

KernelDeimos commented on src/backend/src/modules/puterai/OpenAICompletionService.js:
```diff
@@ -52,15 +52,53 @@ class OpenAICompletionService extends BaseService {
-            this.config?.openai?.secret_key ??
-            this.global_config.openai?.secret_key;
 
+    //Old code 
+    // async _init () {
+    //     const sk_key =
+    //         this.config?.openai?.secret_key ??
+    //         this.global_config.openai?.secret_key;
+
+    //     this.openai = new this.modules.openai.OpenAI({
+    //         apiKey: sk_key
+    //     });
+
+    //     const svc_aiChat = this.services.get('ai-chat');
+    //     svc_aiChat.register_provider({
+    //         service_name: this.service_name,
+    //         alias: true,
+    //     });
+    // }
+
```
Since we use git, we can just remove old code instead of commenting it out.

```suggestion
```

KernelDeimos commented on src/backend/src/modules/puterai/doc/ai-services-config.md:
```diff
@@ -0,0 +1,16 @@
+    "openai": {
+      "apiKey": "sk-abcdefg..."
+    },
+    "other-ai-service": {
```
Maybe we should list what other services are available as well

KernelDeimos reviewed (commented):
Looks good, just two minor comments - I'll merge in 24 hours whether these are addressed or not.

anuja12mishra commented:
@KernelDeimos Sir, I've added an example list of available AI services in the documentation (ai-services-config.md) and removed the comment in OpenAICompletionService.js. Let me know if any further changes are needed!

KernelDeimos commented:
Merged, thanks!
